### ISSUE TITLE
Assertion `result == rr_page_start()' failed to hold. (rr /bin/true)

### ISSUE THREAD

**Issue by fmarier** [mozilla/rr#1537]
Here's what I get when I try to record the execution of `/bin/true` using both `master` and `3.2.0` on Debian jessie (Linux 3.16.0-4-amd64 #1 SMP Debian 3.16.7-ckt11-1+deb8u3 (2015-08-04) x86_64):

```
~/devel/rr-3.2.0/obj$ bin/rr /bin/true
rr: Saving the execution of `/bin/true' to trace directory `/home/francois/.rr/true-2'.
[FATAL /home/francois/devel/rr-3.2.0/src/AddressSpace.cc:359:map_rr_page() errno: 0 'Success'] 
 (task 25918 (rec:25918) at time 15)
 ->  Assertion `result == rr_page_start()' failed to hold. 
Launch gdb with
  gdb /bin/true
and attach to the rr debug server with:
  target remote :25918
```

**Comment by rocallahan** [comment]
Send me email with SSH access? I'll send you my key.

**Comment by fmarier** [comment]
It's all setup (I emailed you the details), just ping me on IRC and I'll turn it on so that you can login.

**Comment by fmarier** [comment]
Rob found that rr can't mmap a file in `/tmp` because I have `/tmp` mounted `noexec` on my laptop.

I can confirm that doing `mount -o remount,exec /tmp` makes the test suite pass.

If rr could:

1. detect this failure and print an error message to that effect
2. provide a way to configure the temporary directory

then I think it would be a good work-around for this issue.

**Comment by rocallahan** [comment]
We can fix this by making the rr page a proper file installed in the same directory as the rr binary.

**Comment by rocallahan** [comment]
Fixed by commit 92f33b1f956cf9cfb452e2e5a6b964f4eb8429fc. Please retest.

**Comment by fmarier** [comment]
Seems to work fine (at least I can record `/bin/true` now). Thanks roc!
